### ISSUE TITLE
Client-side false positive CIRC_MAX_DROPPED_CELLS

### ISSUE THREAD

**Issue by asn-d6** [mikeperry-tor/vanguards#3]
I got this on my Tor client while visiting some hsv3 onions today. I don't think it's legit:
```
NOTICE[Tue May 29 13:33:31 2018]: Circ 30 exceeded CIRC_MAX_DROPPED_BYTES_PERCENT: 0.162515998058 > 2.5. Total: 906360.0, dropped: 152388.0
NOTICE[Tue May 29 13:33:31 2018]: We force-closed circuit 30
```
and then
```
NOTICE[Tue May 29 13:40:23 2018]: Circ 32 exceeded CIRC_MAX_DROPPED_BYTES_PERCENT: 0.0316226845097 > 2.5. Total: 2878440.0, dropped: 96114.0
NOTICE[Tue May 29 13:40:23 2018]: We force-closed circuit 32
```

I see that for clients we do it in notice severity because of potential false positives, but should we also close the circuit?

Also, the inequality in the log msg is a bit funky (0.16 > 2.5).

**Comment by mikeperry-tor** [comment]
Ok I fixed the funky non-percent display.

Wrt closing the circuit -- hrmm.. The way this happens is that a client decides to hang up on a stream while data is still arriving.. Presumably this is due to navigating away or otherwise stopping the page load. In those cases, it seems OK to close the circuit.. If there are other cases, we probably don't want to though..

Let me know if you notice these coinciding with broken page loads or anything like that.

**Comment by mikeperry-tor** [comment]
Idea: bandguards could listen for STREAM events (which only get delivered for client-side onion service streams) and if any are closed with REASON=REQUESTED (or whatever the client-side close REASON field is.. let's hope it's informative), then we could suppress this log line and not kill the circuit.

**Comment by mikeperry-tor** [comment]
So close... The REMOTE_REASON field is supposed to be set in connection_edge_process_relay_cell() if the other side sends the END, but it appears that either that is not happening, or HTTP clients have enough info to hangup early most of the time. :/

**Comment by mikeperry-tor** [comment]
Better idea: https://github.com/torproject/tor/pull/262 (Tor patch for https://trac.torproject.org/projects/tor/ticket/25573).

I have tested and verified that that patch eliminates these instances.

**Comment by mikeperry-tor** [comment]
Fixed in 0.2.3 and 0.3.0
